### PR TITLE
HPCC-14424 Memory stats not all correct

### PR DESCRIPTION
**Pull request by mckellyln**
NOTE: this PR uses some functionality from HPCC-15416 (https://github.com/hpcc-systems/HPCC-Platform/pull/8692) and thus is marked as blocked by that.

@jakesmith please review.

Signed-off-by: Mark Kelly mark.kelly@lexisnexis.com

### CODE REVIEW

**Comment by mckellyln** [comment]
NOTE: this PR is uses some functionality from and is thus blocked by HPCC-15416 (https://github.com/hpcc-systems/HPCC-Platform/pull/8692)

File: system/jlib/jdebug.cpp [hunk 1]
```diff
@@ -947,20 +952,20 @@ memsize_t getMapInfo(const char *type)
         bool skipline = true;
         if ( !strcmp(type, "heap") ) // 'general' heap includes anon mmapped + sbrk
         {
-            if ( strstr(ln, "[heap]") || (!strstr(ln, " /") && !strstr(ln, " [")) )
+            if ( strstr(ln, "[heap") || (!strstr(ln, " /") && !strstr(ln, " [")) )
```
Comment (jakesmith) [inline system/jlib/jdebug.cpp]: Not sure why this and similar diffs to "[stack]"->"[stack" are showing..
Does it need rebasing onto commit from HPCC-15416 ?

File: system/jlib/jdebug.cpp [hunk 2]
```diff
@@ -1118,28 +1155,17 @@ class CInt64fix
...
-    fixarena.set(mi.arena);
-
-    __int64 sbrkmem = fixuordblks.get()+fixusmblks.get();
-    __int64 mmapmem = fixhblkhd.get();
-    __int64 arena =  fixarena.get();
-    __int64 total = mmapmem+sbrkmem;
+    __int64 total = getMapInfo("heap");
+    __int64 sbrkmem = getMapInfo("sbrk");
+    __int64 mmapmem = total - sbrkmem;
+    __int64 proctot = getVMInfo("VmData");
```
Comment (jakesmith) [inline system/jlib/jdebug.cpp]: Would 'virttot' be a better name?

**Comment by jakesmith** [comment]
@mckellyln - looks good, 1 minor comment and question about base commit.

File: system/jlib/jdebug.cpp (hunk 2 above)
Comment (mckellyln) [inline system/jlib/jdebug.cpp]: ok, sure.  Changed.

File: system/jlib/jdebug.cpp (hunk 1 above)
Comment (mckellyln) [inline system/jlib/jdebug.cpp]: It should be "[stack" and "[heap" as there can be more than one of these and they have format "[stack:<tid>]"
I'll make sure its correct in the other PR also.

**Comment by mckellyln** [comment]
I made the minor change and will address your comments in the other commit.

**Comment by mckellyln** [comment]
@jakesmith both PRs updated as per your comments.  Thanks.

**Comment by jakesmith** [comment]
@mckellyln - there are still diffs showing in the 2nd commit, that appear as changes in code in the 1st commit.
HPCC-15416's commit is SHA = ce8edb6, but base commit here is different ( cc90309 )
Looks like needs rebase.

**Comment by mckellyln** [comment]
@jakesmith squashed and rebased

**Comment by mckellyln** [comment]
closing and redoing to get history aligned

**Pull request closed without merging**